frontend/controller/actions/group.js

```diff
@@ -512,6 +514,13 @@ export default (sbp('sbp/selectors/register', {
           })
         }
 
+        // NOTE: sync identity contracts which are out of sync after joining group
+        const missingIDs = (await Promise.all(
+          Object.keys(state.profiles)
+            .map(username => sbp('namespace/lookup', username))
+        )).filter(id => !rootState[id] && !sbp('chelonia/contract/isSyncing', id))
+        await sbp('chelonia/contract/sync', missingIDs)
```

Comment:
I recommend wrapping this in an `if` to check if `missingIDs.length > 0`, and then you can also add a line to: `console.info('found unsynced identity contracts to sync:', missingIDs)`

Reply:
Got it.